text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1003 @@
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This section aims to justify all of the decisions made in the proposed design
+from [*Proposal*][proposal] and discuss why alternatives were not chosen.
+
+## Why put `build-std` in the Cargo config?
+[rationale-build-std-in-config]: #why-put-build-std-in-the-cargo-config
+
+There are various alternatives to putting `build-std` in the Cargo configuration:
+
+1. Cargo could continue to use an explicit command-line flag to enable
+   build-std, such as the current `-Zbuild-std` (stabilised as `--build-std`).
+
+   This approach is proven to work, as per the current unstable implementation,
+   but has a poor user experience, requiring an extra argument to every
+   invocation of Cargo with almost every subcommand of Cargo.
+
+   However, this approach does not lend itself to use with other future and
+   current Cargo features. Additional flags would be required to enable Cargo
+   features (like today's `-Zbuild-std-features`) and would still necessarily be
+   less fine-grained than being able to enable features on individual standard
+   library crates. Similarly for public/private dependencies or customising the
+   profile for the standard library crates.
+
+2. build-std could be enabled or disabled in the `Cargo.toml`. However, under
+   which conditions the standard library is rebuilt is better determined by the
+   user of Cargo, rather than the package being built.
+
+   A user may want to never rebuild the standard library so as to avoid
+   invalidating the guarantees of their qualified toolchain, or may want to
+   rebuild unconditionally to further optimise the standard library for their
+   known deployment platform, or may only want to rebuild as necessary to ensure
+   the build will succeed. All of these rationale can apply to the same crate in
+   different circumstances, so it doesn't make sense for a crate to decide this
+   once in its `Cargo.toml`.
+
+   It would be a waste of resources if a dependency declared that it must always
+   rebuild the standard library when the pre-built crate would be sufficient and
+   this could not be overridden. It is also unclear how to aggregate different
+   configurations of the `build-std` key from different crates in the dependency
+   graph into a single value.
+
+While using `build-std` key in the Cargo configuration shares some of the
+downsides of using an explicit flag - not having a natural extension point for
+other Cargo options exposed to dependencies - [Stage 1b][stage1b] addresses
+these concerns.
+
+↩ [*Proposal*][proposal]
+
+## Why accept `off` as a value for `build-std`?
+[rationale-build-std-off]: #why-accept-off-as-a-value-for-build-std
+
+While not a default value, the user can specify `off` if they prefer which will
+never rebuild the standard library. rustc will still return an error when the
+user's target-modifiers do not match the pre-built standard library.
```

Comment:
Within this Step, `"off"` is listed as the default but this section says its not

Reply:
Must have forgotten to update this from previous iterations, will fix